src/lib/DepositController.js

```diff
@@ -200,6 +183,13 @@ export class DepositController {
       draft = response.data;
     }
 
+    // We have to save draft before we upload files, because files might have
+    // been disabled and we need to re-enable them first. We do it
+    // "stealthily" (not using saveDraft) so as to provide a nice UX.
+    const recordSerializer = store.config.recordSerializer;
+    let payload = recordSerializer.serialize(draft);
+    await this.apiClient.save(payload);
```

Comment:
yeah... we end up having to do this. We probably could be smarter about it and check if there are files already to save some calls. We might also want to take a step back when we have time to refactor some things and make it better (the checkbox could be part of formik and then we could compare against its value and the record in the redux store...)